config/src/main/java/com/typesafe/config/impl/ConfigNodeRoot.java

```diff
@@ -0,0 +1,38 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigSyntax;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+final class ConfigNodeRoot extends ConfigNodeComplexValue {
+    ConfigNodeRoot(Collection<AbstractConfigNode> children) {
+        super(children);
+    }
+
+    protected ConfigNodeComplexValue value() {
+        for (AbstractConfigNode node : children) {
+            if (node instanceof ConfigNodeComplexValue) {
+                return (ConfigNodeComplexValue)node;
+            }
+        }
+        throw new ConfigException.BugOrBroken("ConfigNodeRoot did not contain a value");
```

Comment:
Okay, wrote some tests, and I've verified that the ConfigDocumentParser will create an empty object in the case of an empty document.